### PR TITLE
feat: extract enricher data from config

### PR DESCRIPTION
**Pull request by lstoppa**

### CODE REVIEW

**Comment by lstoppa** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by lstoppa** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by lstoppa** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by lstoppa** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by lstoppa** [comment]
I have read the Code of Conduct and I hereby accept the Terms

File: splunk_connect_for_snmp_poller/manager/static/mib_enricher.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/static/mib_enricher.py:1]: We have duplicated license here

File: splunk_connect_for_snmp_poller/manager/poller.py [hunk 2]
```diff
@@ -194,4 +194,6 @@ def scheduled_task(host, version, community, profile, server_config, splunk_inde
         f"Executing scheduled_task for {host} version={version} community={community} profile={profile}"
     )
 
+    # snmp_polling(host, version, community, profile, server_config, splunk_indexes)
     snmp_polling.delay(host, version, community, profile, server_config, splunk_indexes)
+    # snmp_polling(host, version, community, profile, server_config, splunk_indexes)
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/poller.py:199]: Let's delete this comment as we already have it

File: splunk_connect_for_snmp_poller/manager/static/mib_enricher.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    ) = extract_dimension_name_and_value(dimension, index)
+                    if dimension_name:
+                        result.append({dimension_name: dimension_value})
+        return result
+
+    def process_one(self, translated_var_bind):
+        if translated_var_bind:
+            metric_name = translated_var_bind[InterfaceMib.METRIC_NAME_KEY]
+            additional_if_mib_dimensions = self.__enrich_if_mib(metric_name)
+            if len(additional_if_mib_dimensions) > 0:
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/static/mib_enricher.py:93]: if additional_if_mib_dimensions is a list then:
```
if additional_if_mib_dimensions:
```
is enough

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]

**Review by weliasz — CHANGES_REQUESTED** [review]
How about adding integration test that will show enriched data in Splunk?

File: splunk_connect_for_snmp_poller/manager/static/__init__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
```
[the remark this answers is not in the document]
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/static/__init__.py:1]: > Python tools out there such as mypy and pytest require empty __init__.py files to interpret the code structure accordingly.

File: splunk_connect_for_snmp_poller/manager/static/mib_enricher.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    return None, None
+
+
+class MibEnricher:
+    def __init__(self, mib_static_data_collection):
+        self._mib_static_data_collection = mib_static_data_collection
+
+    def __enrich_if_mib(self, metric_name):
+        result = []
+        if metric_name and metric_name.startswith(InterfaceMib.IF_MIB_METRIC_SUFFIX):
```
[the remark this answers is not in the document]
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/static/mib_enricher.py]: Changed it

**Review by weliasz — APPROVED** [review]

**Review by weliasz — APPROVED** [review]

**Review by omrozowicz-splunk — APPROVED** [review]

**Review by weliasz — APPROVED** [review]

**Pull request merged**